adoptingerlang.org

```diff
@@ -3444,43 +3446,94 @@ initContainers:
+connection to a specific port.
+
+While kubernetes will restart any failed container, there are some times where a
+process becomes unresponsive, such as deadlocks. A =livenessProbe= can be
+used for such cases, where stopping to respond prompts a restart. For this
+reason if a =livenessProbe= is included it should be kept extremely simple. It is
+more often than not safer to leave out the =livenessProbe= entirely and rely on
+the node crashes, metric based alerting and auto-scaling.
+
+One part of auto-scaling is based on the =readinessProbe=. The =readinessProbe=
+tells Kubernetes the =Pod= is ready to receive traffic. When a =Pod= becomes =unready=
+because a container is failing the =readinessProbe=, it is removed from the
+=Service= resulting in traffic being directed at the remaining =Pods=, increasing
+their load, then a new =Pod= will be started.
+
+Not using the =livenessProbe= for this purpose means the containers are still
+available, since they aren't killed, and the reason for failing the check can be
+examined. This may mean attaching a shell if it isn't completely frozen or
+forcing a crash dump to be written. In a very resource constrained environment
+you may prefer to have the =livenessProbe= in order to have deadlocked containers
+and their =Pods= cleaned up right away, instead of waiting for manual intervention
+and the auto-scaler to scale down after load falls.
+
+Another use for the =readinessProbe= is if the application has to periodically do work
+where you don't want to be handling requests, or have to do some sort of
+maintenance, it can return a =503= status response for readiness causing it to be
+removed from the =Service= backends.
+
+In the =Pod= configuration nothing special is needed to have a graceful shutdown,
+whether it is during a deploy, =livenessProbe= failure or scale down. Erlang will
+call =init:stop()= when =SIGTERM= is sent by Kubernetes to signal shutdown. As
+discussed in [[Releases]], each Applications will be stopped in the reverse they
+were started and each Application's supervision tree terminates children in
+reverse order.
+
+By default the process can wait up to 30 seconds (configurable with the
+=terminationGracePeriodSeconds= =PodSpec= option) before Kubernetes sends a =SIGKILL=
+signal that will force the process to terminate.
+
+An important note about deployments and =readinessProbe= is the behaviour if a
+=readinessProbe= never passes. During deployment if the =Pod= never passes its
+=readinessProbe= then, depending on the deployment strategy, =Pods= from the
+previous version of the deployment that are =Ready= may remain until they do. This
+is covered more in the follow section [[Rolling Deployments]].
+
+In the =service_discovery= project only =readinessProbe= is defined for the purpose
+of removing the =Pod= from =Service= backends without killing the container:
 
 #+BEGIN_SRC yaml
-livenessProbe:
+readinessProbe:
   httpGet:
-    path: /healthz
+    path: /ready
```

Comment:
might be interesting to eventually aim for `/health/<type>` but we're good to go here.